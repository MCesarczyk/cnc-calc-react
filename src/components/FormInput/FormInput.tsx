import { FieldWrapper } from '../FieldWrapper';
import { Input } from '../Input';
import { TextSub } from '../TextSub/TextSub';
import { Label } from '../Label';
import { ChangeEvent } from 'react';

interface FormInputProps {
  name: string;
  sub?: string;
  unit?: string;
  inputRef?: any;
  value: string;
  min?: number;
  step?: number;
  placeholder?: string;
  disabled?: boolean;
  readOnly?: boolean;
  autoFocus?: boolean;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
}

export const FormInput = ({
  name,
  sub,
  unit,
  inputRef,
  value,
  min,
  step,
  placeholder,
  disabled,
  readOnly,
  autoFocus,
  onChange,
}: FormInputProps) => {
  return (
    <FieldWrapper>
      <Label>
        <label>
          {name}
          <TextSub>{sub}</TextSub>
          {unit}
        </label>
      </Label>
      <Input
        ref={inputRef}
        value={value}
        type="number"
        min={min}
        step={step}
        placeholder={placeholder}
        required
        disabled={disabled}
        readOnly={readOnly}
        autoFocus={autoFocus}
        onChange={onChange}
      />
    </FieldWrapper>
  );
};
