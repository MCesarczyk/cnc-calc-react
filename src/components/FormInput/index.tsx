import { FieldWrapper } from "../FieldWrapper";
import { Input } from "../Input";
import { TextSub } from "../TextSub";
import { Label } from "../Label";

interface FormInputProps {
  name: string;
  sub?: string;
  unit?: string;
  inputRef?: any;
  value?: any;
  min?: number;
  step?: number;
  placeholder?: string;
  disabled?: boolean;
  readOnly?: boolean;
  autoFocus?: boolean;
  onChange?: any;
}

const FormInput = ({
  name,
  sub,
  unit,
  inputRef,
  value,
  min,
  step,
  placeholder,
  disabled,
  readOnly,
  autoFocus,
  onChange
}:FormInputProps) => {
  return (
    <FieldWrapper>
      <Label>
        <label>
          {name}
          <TextSub>
            {sub}
          </TextSub>
          {unit}
        </label>
      </Label>
      <Input
        ref={inputRef}
        value={value}
        type="number"
        min={min}
        step={step}
        placeholder={placeholder}
        required
        disabled={disabled}
        readOnly={readOnly}
        autoFocus={autoFocus}
        onChange={onChange}
      />
    </FieldWrapper>
  )
};

export default FormInput;