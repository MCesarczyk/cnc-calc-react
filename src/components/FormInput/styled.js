import styled from "styled-components";

export const InputLabelText = styled.span`
    text-align: right;

    @media (max-width: 480px) {
        text-align: center;
    }
`;