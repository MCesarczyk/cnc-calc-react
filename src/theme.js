export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}