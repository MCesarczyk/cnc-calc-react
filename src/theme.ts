export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
