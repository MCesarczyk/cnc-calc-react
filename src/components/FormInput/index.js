import { Input } from "../Input";
import { TextSub } from "../TextSub";
import { InputLabelText, FieldWrapper } from "./styled";

const FormInput = ({
  name,
  sub,
  unit,
  inputRef,
  value,
  type,
  min,
  step,
  placeholder,
  required,
  disabled,
  readOnly,
  autoFocus,
  onChange
}) => {
  return (
    <FieldWrapper>
      <InputLabelText>
        <label>
          {name}
          <TextSub>
            {sub}
          </TextSub>
          {unit}
        </label>
      </InputLabelText>
      <Input
        ref={inputRef}
        value={value}
        type={type}
        min={min}
        step={step}
        placeholder={placeholder}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        autoFocus={autoFocus}
        onChange={onChange}
      />
    </FieldWrapper>
  )
};

export default FormInput;