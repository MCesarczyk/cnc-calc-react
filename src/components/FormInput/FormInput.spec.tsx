import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { theme } from 'theme';
import { FormInput } from './FormInput';

const setup = () =>
  render(
    <ThemeProvider theme={theme}>
      <FormInput inputName="" name={''} value={''} onChange={() => {}} />
    </ThemeProvider>,
  );

test('Input component should be empty initially', () => {
  const input = setup().getByRole('spinbutton');
  // @ts-ignore
  expect(input.value).toBe('');
});

test('Input component should display changed value of state variable', () => {
  const input = setup().getByRole('spinbutton');
  fireEvent.change(input, { target: { value: '123' } });
  // @ts-ignore
  expect(input.value).toBe('');
});

test("Input component shouldn't allow to input letters", () => {
  const input = setup().getByRole('spinbutton');
  fireEvent.change(input, { target: { value: 'Qwerty' } });
  // @ts-ignore
  expect(input.value).toBe('');
});

test('FormInput component rendering properly with styles', () => {
  const input = renderer.create(
    <ThemeProvider theme={theme}>
      <FormInput inputName="" name={''} value={''} onChange={() => {}} />
    </ThemeProvider>,
  );

  let tree = input.toJSON();

  expect(tree).toMatchSnapshot();
});
