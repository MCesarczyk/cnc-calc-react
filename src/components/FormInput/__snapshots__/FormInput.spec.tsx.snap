// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component rendering properly with styles 1`] = `
.c0 {
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;
}

.c2 {
  vertical-align: sub;
  font-size: 0.7em;
  margin-right: 0.15em;
}

.c1 {
  text-align: right;
}

.c3 {
  border: solid 1px #1e3246;
  border-radius: 5px;
  padding: 5px;
}

.c3:invalid {
  background-color: beige;
}

.c3:disabled {
  background-color: #ddd;
}

@media (max-width: 480px) {
  .c0 {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
}

@media (max-width: 480px) {
  .c1 {
    align-self: flex-start;
  }
}

@media (max-width: 480px) {
  .c3 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <label
      htmlFor=""
    >
      <span
        className="c2"
      />
    </label>
  </label>
  <input
    className="c3"
    id=""
    name=""
    onChange={[Function]}
    required={true}
    type="number"
    value=""
  />
</div>
`;
